test/units/modules/network/aci/test_aci_rest.py

```diff
@@ -0,0 +1,299 @@
+            u'imdata': [{
+                u'error': {
+                    u'attributes': {
+                        u'code': u'401',
+                        u'text': u'Username or password is incorrect - FAILED local authentication',
+                    },
+                },
+            }],
+            u'totalCount': '1',
+        }
+
+        json_response = '{"totalCount":"1","imdata":[{"error":{"attributes":{"code":"401","text":"Username or password is incorrect - FAILED local authentication"}}}]}'  # NOQA
+        json_result = aci_response(json_response, 'json')
+
+        # Python 2.7+ is needed for xmljson
+        if sys.version_info < (2, 7):
+            return
+
+        xml_response = '''<?xml version="1.0" encoding="UTF-8"?><imdata totalCount="1">
+        <error code="401" text="Username or password is incorrect - FAILED local authentication"/>
+        </imdata>
+        '''
+        xml_result = aci_response(xml_response, 'xml')
+
+        self.assertEqual(expected_result, json_result)
+        self.assertEqual(json_result, xml_result)
+
+    def test_empty_response(self):
+        self.maxDiffi = None
+
+        if sys.version_info < (3, 0):
+            expected_json_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as JSON, see 'raw' output. No JSON object could be decoded",
+                'raw': '',
+            }
+
+        else:
+            expected_json_result = {
+                u'error_code': -1,
+                u'error_text': u"Unable to parse output as JSON, see 'raw' output. Expecting value: line 1 column 1 (char 0)",
+                u'raw': u'',
+            }
+
+        json_response = ''
+        json_result = aci_response(json_response, 'json')
+        self.assertEqual(expected_json_result, json_result)
+
+        # Python 2.7+ is needed for xmljson
+        if sys.version_info < (2, 7):
+            return
+
+        elif etree.LXML_VERSION < (3, 3, 0, 0):
+            expected_xml_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as XML, see 'raw' output. None",
+                'raw': '',
+            }
+
+        elif sys.version_info < (3, 0):
+            expected_xml_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as XML, see 'raw' output. None (line 0)",
+                'raw': '',
+            }
+
+        else:
+            expected_xml_result = {
+                u'error_code': -1,
+                u'error_text': u"Unable to parse output as XML, see 'raw' output. None (line 0)",
+                u'raw': u'',
+            }
+
+        xml_response = ''
+        xml_result = aci_response(xml_response, 'xml')
+        self.assertEqual(expected_xml_result, xml_result)
+
+    def test_invalid_response(self):
+        self.maxDiff = None
+
+        if sys.version_info < (2, 7):
```

Comment:
Another test that I would move into version specific classes with pytest skipifs